File: 3794.reverse-string-prefix.java
/*
 * @lc app=leetcode id=3794 lang=java
 *
 * [3794] Reverse String Prefix
 */

// @lc code=start
class Solution {
    public String reversePrefix(String s, int k) {
        char[] arr = s.toCharArray();
        int left = 0;
        int right = k - 1;
        while (left <= right) {
            char temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
        return new String(arr);
    }
}
// @lc code=end

File: 3794.reverse-string-prefix.py
#
# @lc app=leetcode id=3794 lang=python
#
# [3794] Reverse String Prefix
#

# @lc code=start
class Solution(object):
    def reversePrefix(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        s = list(s)
        left = 0
        right = k - 1
        while left <= right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1
        return ''.join(s)
# @lc code=end

